pytest_asyncio/plugin.py

```diff
@@ -249,14 +248,38 @@ def _add_kwargs(
     return ret
 
 
-def _wrap_asyncgen(func: Callable[..., AsyncIterator[_R]]) -> Callable[..., _R]:
-    @functools.wraps(func)
+def _perhaps_rebind_fixture_func(
+    func: _T, instance: Optional[Any], unittest: bool
```

Comment:
The type checker complains that `_T` doesn't have the `__func__`, `__self__`, and `__get__` attributes. We need to address this before merging the PR.

I'm not aware of a proper type bound for `_T` that would fix the issue (or even a type that expresses a bound method). Ideas are welcome, but IMHO it would be acceptable to ignore the warnings inside of `_perhaps_rebind_fixture_func`.